base/acme/src/main/java/org/dogtagpki/acme/server/ACMENewOrderService.java

```diff
@@ -83,13 +84,12 @@ public Response createNewOrder(JWS jws) throws Exception {
             // identifier value.  The returned authorization MUST include the
             // optional "wildcard" field, with a value of true.
 
-            boolean wildcard;
-            if ("dns".equals(type) && value.startsWith("*.")) {
-                wildcard = true;
-                value = value.substring(2); // remove *. prefix
-
-            } else {
-                wildcard = false;
+            boolean wildcard = false;
+            if ("dns".equals(type)) {
+                wildcard = engine.getPolicy().validateWildcard(value);
+                if (wildcard && value.startsWith("*.")) {
```

Comment:
If `wildcard == true`, it means that the `value` contains a valid wildcard, so we don't really need to check `value.startsWith("*.")` again.

But if we remove it the code might look a little weird. What do you think?

Reply:
I left that because we had a discussion on IRC and you mentioned possibly supporting "x.*.name" wildcards later. Obviously we wouldn't remove a starting "*." prefix because it wouldn't exist in that case. 

That said, I'm fine with whatever.